content/blog/2022-08-08-Mid-Term-Evaluation-Updates-PawBud.md

```diff
@@ -0,0 +1,87 @@
+            'template_hook': 'getJingleTemplate'
+        }
+        context.model.messages.create(attrs);
+    }
+```
+
+I borrowed this code from a single file called "utils.js" which as the name suggests contains all the miscellaneous functions of a plugin that are needed for it to function. It doesn't take much thinking to realize how big the problem becomes once the codebase is large enough. I must say it's thanks to my mentor, I was able to differentiate the two sides, and i am now able to move on Though, I still get confused but fortunately, I feel like its becoming more infrequent.
+
+Ohh, and by the way, the [create](https://backbonejs.org/#Collection-create) function creates an instance of the message model, and if you didn't understand this statement, I highly recommend going through the [backbone.js documentation](https://backbonejs.org/#).
+
+So Lessons to Learn:-
+* Comment or possibly document your code from day one.
+* Don't keep random variable names.
+* Speak up to your mentor!! Period. I can assure you a lot of JC's time could have been saved had I spoken up about my thought process sooner.
+
+Ahemmm..... let's move on now.
+
+### Message Chat History Rendering for Jingle
+
+Now, when you get a missed call, you usually get a timestamp. It is no different in a jingle call, but I do need to have a discussion regarding it's actual implementation with JC. The actual message content should be pretty straightforward and if you want to see how I implemented/am implementing it, I will straight up point towards [JC's explanation of it](https://github.com/conversejs/converse.js/issues/447#issuecomment-1198217084).
+
+### Non-Technical Corner of this blog
+
+Hmmmm enough about the project, now for some unnecessary life updates. I will be leaving for my master's program in a week's time. My schedule is packed with various things right now. Going to a new country on your own is scary for the first time, especially when you know that there is no one who can help you there but GSoC seems like a perfect distraction from the nervousness. JC seems to be doing fine, speaking of whom, did I mention he took a hiatus last week? I really hope he enjoyed his time off. There were nights in the last week, which made me wonder if I did actually stress him out or something, but I am pretty sure I am just overthinking it.I value his patience out of all things and I am not joking when I say he has a lot of it. I mean, if I as a mentor would meet another PawBud, I would have probably quit by now and to be honest, that's one of the million reasons why I am not a mentor right now. I feel so embarrassed about the little things JC has to repeat every time because I was distracted or too lazy to note them down. I hate inconsistency and love to be direct about things, so if you have made it this far, I would like to say that there is a good chance that this project might end with the implementation of XEP-0353 given my slow pace.
```

Comment:
```suggestion
Hmmmm enough about the project, now for some unnecessary life updates. I will be leaving for my master's program in a week's time. My schedule is packed with various things right now. Going to a new country on your own is scary for the first time, especially when you know that there is no one who can help you there but GSoC seems like a perfect distraction from the nervousness. JC seems to be doing fine, speaking of whom, did I mention he took a hiatus last week? I really hope he enjoyed his time off. There were nights in the last week, which made me wonder if I did actually stress him out or something, but I am pretty sure I am just overthinking it. I value his patience out of all things and I am not joking when I say he has a lot of it. I mean, if I as a mentor would meet another PawBud, I would have probably quit by now and to be honest, that's one of the million reasons why I am not a mentor right now. I feel so embarrassed about the little things JC has to repeat every time because I was distracted or too lazy to note them down. I hate inconsistency and love to be direct about things, so if you have made it this far, I would like to say that there is a good chance that this project might end with the implementation of XEP-0353 given my slow pace.
```